### ISSUE TITLE
Type RequestTransformFunction should be RequestTransformFunction

### ISSUE THREAD

**Issue by chrisl777** [alex3165/react-mapbox-gl#666]
While building my project, I noticed that the RequestTransformFunction type is incorrectly named versus @types/mapbox-gl. See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c6f73c8050c16ceb3a9ac23df466361e9e22a4a1/types/mapbox-gl/index.d.ts

The error message when I compile with TypeScript: 

```
/node_modules/react-mapbox-gl/lib/map.d.ts
(82,13): Subsequent property declarations must have the same type.  Property 'transformRequest' must be of type 'TransformRequestFunction | undefined', but here has type 'RequestTransformFunction | undefined'.
```

**Comment by Alex1100** [comment]
Is there a fix for this?

**Comment by henrinormak** [comment]
Would also be interested in a fix for this, seems like a fairly serious problem (in essence the library is not useable).

**Comment by amwill04** [comment]
The `types` seem to be all over the place. Having massive problems.

First  after installing `@types/supercluster` you get `"@types/supercluster" has no exported member 'Supercluster'` so to fix that set version to `@types/supercluster@3.0.2`.

Then have this problem which I solved by basically changing the `type` name from `RequestTransformFunction` to `TransformRequestFunction` in `map.d.ts`. 

No idea if this is an ideal fix or what problems I will run into - but it compiles.

**Comment by danvk** [comment]
The only workaround I've found is to downgrade to TypeScript 3.1 or earlier.

**Comment by amwill04** [comment]
@danvk I’ve personally just forked it and fixed the types. More than happy to submit a PR

**Comment by danvk** [comment]
@amwill04 @pducks32 beat you to it: https://github.com/alex3165/react-mapbox-gl/pull/672

**Comment by amwill04** [comment]
@danvk ha, fair play. Makes my life a hell of a lot easier.

**Comment by mklopets** [comment]
Fix was released in `v4.2.4`
